### PR TITLE
Blog Post: Deploy Bitbucket to AWS in Five Minutes

### PR DESCRIPTION
**Pull request by ianmalott**
## Release Playbook
- [ ] Rebase against master
- [ ] Code review
- [ ] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

File: _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+2. In that directory, create a docker-compose.yml with the following contents.
+    <pre>
+    version: '2'
+    services:
+      web:
+        image: atlassian/bitbucket-server
+        ports:
+          - 80:7990</pre>
+3. Run `convox apps create bitbucket-server` to prepare a new app.
+4. Run `convox scale --memory=1024` to allocate more than the default 256MB.
```
Comment (mattmanning) [inline _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md]: Does this `convox scale` command actually work? I'd think you'd need the `web` process deployed first and to do `convox scale web`?

An alternative is to use a `mem_limit` entry in docker-compose.yml

File: _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+1. [Install the Convox CLI](https://convox.com/docs/installation/).
+2. Create a new directory called 'bitbucket-server'.
+3. In that directory, create a docker-compose.yml with the following contents.
+    <pre>
+    version: '2'
+    services:
+      web:
+        image: atlassian/bitbucket-server
+        ports:
+          - 80:7990</pre>
```
Comment (mattmanning) [inline _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md]: If we add

```
    volumes:
      - /var/atlassian/application-data/bitbucket
```

we can get data persistence. I just tried that out and it works locally.

Comment (ianmalott) [inline _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md]: Do you think that's a win for this post? If someone needs to do more than kick the tires, they'll probably want to set up an external database anyway, and that's beyond the scope of this writeup I think.

Then again, I could add a "Taking it further section" that shares that volumes config, links to convox.com/docs/volumes, and talks about setting up an external database if someone wants to go beyond trial mode. What do you think?

Comment (ianmalott) [inline _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md]: https://confluence.atlassian.com/bitbucketserver/connecting-bitbucket-server-to-an-external-database-776640378.html

Comment (ianmalott) [inline _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md]: Guess I could try to get a deployed Bitbucket Server working with a Convox Postgres resource. Looks like it'd be pretty doable:

screen shot 2017-03-06 at 4 57 36 pm

File: _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md (hunk 1 above)
Comment (ianmalott) [inline _posts/2017-03-03-deploy-bitbucket-to-AWS-in-five-minutes.md]: Good save. Thank you! Went with the `mem_limit` entry to save time (can't run `convox scale ...` while the stack is already updating) and an extra step.

**Comment by mattmanning** [comment]
Nice update to the conclusion.

**Pull request merged**